staging/src/k8s.io/client-go/util/workqueue/default_rate_limiters.go

```diff
@@ -174,6 +174,7 @@ func (r *ItemFastSlowRateLimiter) Forget(item interface{}) {
 // were separately delayed a longer time.
 type MaxOfRateLimiter struct {
```

Comment:
I think this can be more cleanly constructed by building a peer struct

```go
type WithMaxWaitRateLimiter struct {
	limiter RateLimiter
	maxDelay time.Duration
}

func WithMaxWait(limiter RateLimiter, maxDelay time.Duration ) RateLimiter {
	return &WithMaxWaitRateLimiter{limiter: limiter, maxDelay: maxDelay}
}
```

Allowing for construction like

```go
WithMaxWait(DefaultControllerRateLimiter(), time)
```

Reply:
ok, i will modify it immediately